enforcer.go

```diff
@@ -537,13 +545,24 @@ func (e *Enforcer) enforce(matcher string, explains *[]string, rvals ...interfac
 	var expression *govaluate.EvaluableExpression
 	hasEval := util.HasEval(expString)
 
-	if hasEval {
+	if !hasEval {
+		var cachedExpression, isPresent = e.expressions.Load(expString)
```

Comment:
This code logic is way too complicated when messed up together, can we extract a wrapper out which will use cache?

Reply:
@hsluoyz do you mean wrap the caching code in a separate func?

Reply:
Code cleanup done, and made the logic easier to understand